### PR TITLE
MSI-757: Add Inventory Index invalidation when Source is enabled or disabled.

### PR DESCRIPTION
**Pull request by p-bystritsky**
### Fixed Issues (if relevant)

1. magento-engcom/msi#757: Add Inventory Index invalidation when Source is enabled or disabled.

### CODE REVIEW

File: app/code/Magento/InventoryIndexer/Plugin/InventoryApi/InvalidateAfterSourceSetEnabledPlugin.php [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+    public function aroundSave(
+        SourceRepositoryInterface $subject,
+        callable $proceed,
+        SourceInterface $source
+    ) {
+        $sourceCode = $source->getSourceCode();
+        $oldStatus = $this->getSourceEnabled->execute($sourceCode);
+        $proceed($source);
+        $status = (int)$source->isEnabled();
+        // Non-strict compassion needed.
```
Comment (naydav) [inline app/code/Magento/InventoryIndexer/Plugin/InventoryApi/InvalidateAfterSourceSetEnabledPlugin.php]: Why? we can use cast

File: app/code/Magento/InventoryIndexer/Test/Integration/Plugin/InventoryApi/InvalidateAfterSourceSetEnabledPluginTest.php [hunk 2]
```diff
@@ -0,0 +1,186 @@
...
+    }
+
+    /**
+     * Contains generic logic for all Indexer invalidation after Source enabled or disabled tests.
+     *
+     * @param string $sourceCode
+     * @param bool $enable
+     * @param bool $expectedValid
+     */
+    private function indexerInvalidationBase(string $sourceCode, bool $enable, bool $expectedValid)
```
Comment (naydav) [inline app/code/Magento/InventoryIndexer/Test/Integration/Plugin/InventoryApi/InvalidateAfterSourceSetEnabledPluginTest.php]: if it is `assert` rename method to `assertWhatIWantToCheck`

File: app/code/Magento/InventoryIndexer/Model/GetSourceEnabled.php [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+namespace Magento\InventoryIndexer\Model;
+
+use Magento\Framework\App\ResourceConnection;
+use Magento\Inventory\Model\ResourceModel\Source;
+use Magento\InventoryApi\Api\Data\SourceInterface;
+
+/**
+ * Get Enabled field value for Source by Source Code.
+ */
+class GetSourceEnabled
```
Comment (naydav) [inline app/code/Magento/InventoryIndexer/Model/GetSourceEnabled.php]: Looks like we can merge classes in one and perform one query
We need this check only in one place

**Review by naydav — CHANGES_REQUESTED** [review]

File: app/code/Magento/InventoryIndexer/Plugin/InventoryApi/InvalidateAfterSourceSetEnabledPlugin.php (hunk 1 above)
Comment (p-bystritsky) [reply app/code/Magento/InventoryIndexer/Plugin/InventoryApi/InvalidateAfterSourceSetEnabledPlugin.php]: @naydav done.

File: app/code/Magento/InventoryIndexer/Test/Integration/Plugin/InventoryApi/InvalidateAfterSourceSetEnabledPluginTest.php (hunk 2 above)
Comment (p-bystritsky) [reply app/code/Magento/InventoryIndexer/Test/Integration/Plugin/InventoryApi/InvalidateAfterSourceSetEnabledPluginTest.php]: @naydav no. We need to run the same test with the different fixtures loaded. This method contains generic logic for all those tests.

File: app/code/Magento/InventoryIndexer/Model/GetInvalidationRequired.php [hunk 4]
```diff
@@ -0,0 +1,66 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+declare(strict_types=1);
+
+namespace Magento\InventoryIndexer\Model;
```
Comment (RomaKis) [inline app/code/Magento/InventoryIndexer/Model/GetInvalidationRequired.php]: Seems this class is better to move to Resource namespace.

Comment (p-bystritsky) [reply app/code/Magento/InventoryIndexer/Model/GetInvalidationRequired.php]: @RomaKis done.

**Pull request merged**